### ISSUE TITLE
Make gnome apps appear automatically in applications menu upon install

### ISSUE THREAD

**Issue by quatquatt** [quatquatt/nixos#42]
Not sure if this is possible, but would certainly be nice. 

Relevant links: 

https://github.com/NixOS/nixpkgs/issues/12757

https://www.reddit.com/r/NixOS/comments/ni4vjd/how_to_update_gnome_application_list_icons_after/

https://www.reddit.com/r/NixOS/comments/ofmzb9/comment/jzlolrc/?share_id=AkmQS3yMYBtGrNocXg5sB

**Comment by radioaddition** [comment]
Per the second link: unfortunately apt+f2 restarting only works on x11, not Wayland. Not sure if that means it's not possible on Wayland or if you just gotta do some cli stuff to get it to work

**Comment by quatquatt** [comment]
> Per the second link: unfortunately alt+f2 restarting only works on x11, not Wayland. Not sure if that means it's not possible on Wayland or if you just gotta do some cli stuff to get it to work

Yeah, that one's stupid and old. Just wanted to put everything I'd found. The other ones seem promising, though.

**Comment by radioaddition** [comment]
Found a link and put it on my issue tracker that may interest you

**Comment by quatquatt** [comment]
> Found a link and put it on my issue tracker that may interest you

Awesome, putting it here for my future reference: 
https://unix.stackexchange.com/questions/12118/how-do-i-refresh-gnome-3-applications/565807#565807

**Comment by quatquatt** [comment]
@radioaddition 
After doing some testing, I'm completely confident this is a home-manager issue. Unfortunately, it means that the solutions with `xdg-desktop-menu forceupdate` and `update-desktop-database ~/.local/share/applications` aren't helpful for us. 

When I install gnome-calendar through `home.packages`, it doesn't appear in the gnome application list. When I then take it out of `home.packages`, and move it into `environment.systemPackages`, it shows up immediately. Any apps I've put in `environment.systemPackages` have appeared in the application list immediately. They don't show up with their desktop icon, but that appears after one *more* rebuild.

That sounds simple. But it gets more complicated. 

Many of my *currently* installed apps seem completely unable to ever appear in the app list. I've tested with Kitty, VSCode, and Obsidian. Kitty and VSCode were installed via a home-manager module, via `programs.programName.enable`. Obsidian was installed via `home.packages` originally. Whether I move these to `pkgs-unstable`, remove and reinstall them, run the commands specified in the stackoverflow post, nothing seems to work. 

This could be a me problem. Being previously on standalone and now installed as a module, with homeConfigurations still used, is absolutely a strange paradigm. I don't trust my machine as being representative of the average case. I plan on reinstalling NixOS to my desktop whenever I get the time, and I think that'll be a much better true test. Maybe the apps will just appear and I'll know it's a Framework Laptop skill issue. 

If you have any luck reproducing any of this, let me know!

**Comment by quatquatt** [comment]
Maybe it's a home-manager standalone issue. After all, I was previously on standalone, and am now on home-manager as a module. As a module, it seems like removing things from `home.packages` and putting them in `environment.systemPackages` makes them immediately appear. But all the currently-installed apps that don't seem to work were installed *before*, when I was on standalone. So maybe there's a standalone bug where somehow the state of the app not appearing in the Applications menu persists. 

(This is a shaky hypothesis with the evidence I have: a real clean test is needed, with someone like you trying to reproduce on standalone, and me trying to reproduce with a clean install on my desktop.)

**Comment by radioaddition** [comment]
Mark as closed with solution: just don't use home.packages

**Comment by radioaddition** [comment]
Oh shit I thought this was my issue tracker XD

**Comment by quatquatt** [comment]
Well, meanwhile, I have *not* solved the issue. When booting into a VM, it appeared like this: 
![Screenshot from 2024-09-24 20-15-34](https://github.com/user-attachments/assets/13bb9523-0726-42c4-a918-455f252fa46e)
This is even *less* apps than I have installed right now. 

Next step is to see what happens when we reinstall.

**Comment by damccull** [comment]
Also experiencing this with some apps. Obsidian is my bane right now.
I have solved it manually for now with a little `ln -s $(pwd)/share/applications/obsidian.desktop ~/.local/share/applications/obsidian.desktop` action, but that's clearly not going to survive the next update I get for this package.

**Comment by llakala** [comment]
> Also experiencing this with some apps. Obsidian is my bane right now.
> I have solved it manually for now with a little `ln -s $(pwd)/share/applications/obsidian.desktop ~/.local/share/applications/obsidian.desktop` action from inside the nix store package directory, but that's clearly not going to survive the next update I get for this package.

Interesting. Do you have Home-Manager installed standalone (or did you ever on the current system since the last reinstall)?

**Comment by damccull** [comment]
My home manager is run as part of the system update, so it's integrated. But I installed obsidian via env.packages rather than through home manager. I keep forgetting to check home-manager lol.
